rdt-open-reader/src/covidIndonesia/java/io/ona/rdt/widget/OneScanCovidRDTBarcodeFactory.java

```diff
@@ -37,10 +41,24 @@
     protected void launchBarcodeScanner(Activity activity, MaterialEditText editText, String barcodeType) {
         InputMethodManager inputManager = (InputMethodManager) activity.getSystemService(Context.INPUT_METHOD_SERVICE);
         inputManager.hideSoftInputFromWindow(editText.getWindowToken(), HIDE_NOT_ALWAYS);
-        Intent intent = new Intent(activity, OneScanActivity.class);
-        intent.putExtra(OneScanActivity.ENABLE_BACK_PRESS, widgetArgs.getJsonObject().optBoolean(OneScanActivity.ENABLE_BACK_PRESS));
-        intent.putExtra(Constants.Config.ENABLE_BATCH_SCAN, widgetArgs.getJsonObject().optBoolean(Constants.Config.ENABLE_BATCH_SCAN));
-        activity.startActivityForResult(intent, BARCODE_REQUEST_CODE);
+
+        Intent scannerIntent = new Intent();
+        scannerIntent.setClassName(OneScanHelper.PACKAGE_NAME, OneScanHelper.ACTIVITY_NAME);
+
+        if (OneScanHelper.isCallable(activity, scannerIntent)) {
+            Intent intent = new Intent(activity, OneScanActivity.class);
+            intent.putExtra(OneScanActivity.ENABLE_BACK_PRESS, widgetArgs.getJsonObject().optBoolean(OneScanActivity.ENABLE_BACK_PRESS));
+            intent.putExtra(Constants.Config.ENABLE_BATCH_SCAN, widgetArgs.getJsonObject().optBoolean(Constants.Config.ENABLE_BATCH_SCAN));
+            activity.startActivityForResult(intent, BARCODE_REQUEST_CODE);
+        } else {
+            AlertDialog.Builder builder = new AlertDialog.Builder(activity);
+            builder.setTitle(R.string.error)
+                    .setMessage(R.string.onescan_is_not_installed)
+                    .setPositiveButton(R.string.ok, (dialog, id) -> activity.onBackPressed());
+
+            AlertDialog dialog = builder.create();
+            dialog.show();
+        }
```

Comment:
https://github.com/onaio/rdt-standard/blob/10e8c169ba5261c70f51f09dc6c36c78b6366ad9/rdt-open-reader/src/covidIndonesia/java/io/ona/rdt/util/OneScanHelper.java#L43 we can add the alertdialog here instead, since we already check for `isCallable`.

Reply:
you can use `Utils.showAlertDialog()` to build the dialog.

Reply:
I did the changes:
![Screenshot 2021-03-16 094414](https://user-images.githubusercontent.com/79955578/111248981-8b1d1c80-863d-11eb-9f48-78c9c3ee5cc6.png)

But the screen not showing as expected:
![Screenshot 2021-03-16 094444](https://user-images.githubusercontent.com/79955578/111248994-91ab9400-863d-11eb-8ddf-58950bcd63dc.png)

it supposed to go back to previous screen.
Can I just put it in OneScanCovidRDTBarcodeFactory?